drivers/bmx055/bmx055.c

```diff
@@ -0,0 +1,241 @@
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_bmx055
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for the BMX055 9-axis sensor
+ *
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stdint.h>
+#include <math.h>
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "assert.h"
+
+#include "bmx055.h"
+#include "bmx055_internal.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define BUS_CLK         (I2C_SPEED_FAST)
+#define BUS             (dev->p.i2c)
+#define ADDR_MAG        (dev->p.addr_mag)
+#define ADDR_ACC        (dev->p.addr_acc)
+#define ADDR_GYRO       (dev->p.addr_gyro)
+#define GYRO_2000_DPS   (2000U)
+#define GYRO_MAX_SCALE  (0x7FFFU)
+
+/**
+ * @brief Array of available range values of accelerometer
+ */
+static const uint8_t acc_ranges[] = {
+    BIT_ACC_RANGE_2G,
+    BIT_ACC_RANGE_4G,
+    BIT_ACC_RANGE_8G,
+    BIT_ACC_RANGE_16G,
+};
+
+int bmx055_init(bmx055_t *dev, const bmx055_params_t *params)
+{
+    assert(dev && params);
+
+    uint8_t tmp;
+
+    memcpy(&dev->p, params, sizeof(bmx055_params_t));
+
+    /* init i2c Bus */
+    if (i2c_init_master(dev->p.i2c, BUS_CLK) != 0) {
+        DEBUG("[bmx055] error: unable to init i2c bus\n");
+        return BMX055_NOBUS;
+    }
+
+    /* bring magnetometer from suspend mode to sleep mode just in case
+     * and try to read magnetometer id
+     * NOTE: this is necessary because the module id is 0x00 in suspend mode */
+    if (i2c_write_reg(BUS, ADDR_MAG, REG_MAG_PWRCTRL, BIT_MAG_PWRCTRL_VAL) != 1) {
+        DEBUG("[bmx055] error: no connection to magnetometer\n");
+        return BMX055_NODEV;
+    }
+    if (i2c_read_reg(BUS, ADDR_MAG, REG_MAG_CHIPID, &tmp) != 1) {
+        DEBUG("[bmx055] error: no connection to magnetometer\n");
+        return BMX055_NODEV;
+    }
+    if (tmp != REG_MAG_CHIPID_VAL) {
+        DEBUG("[bmx055] error: no connection to magnetometer\n");
+        return BMX055_NODEV;
+    }
+
+    /* try to read accelerometer id */
+    if (i2c_read_reg(BUS, ADDR_ACC, REG_ACC_CHIPID, &tmp) != 1) {
+        DEBUG("[bmx055] error: no connection to accelerometer\n");
+        return BMX055_NODEV;
+    }
+    if (tmp != REG_ACC_CHIPID_VAL) {
+        DEBUG("[bmx055] error: no connection to accelerometer\n");
+        return BMX055_NODEV;
+    }
+
+    /* try to read gyroscope id */
+    if (i2c_read_reg(BUS, ADDR_GYRO, REG_GYRO_CHIPID, &tmp) != 1) {
+        DEBUG("[bmx055] error: no connection to gyroscope\n");
+        return BMX055_NODEV;
+    }
+    if (tmp != REG_GYRO_CHIPID_VAL) {
+        DEBUG("[bmx055] error: no connection to gyroscope\n");
+        return BMX055_NODEV;
+    }
+
+    /* Init Magnetometer
+     *
+     * set magnetometer to normal mode (Bits 1 & 2 = 0x00)
+     * and set magnetometer sample rate (Bits 3 to 5) */
```

Comment:
minor nit, place `*/` on new line, i.e.

```
 * and set magnetometer sample rate (Bits 3 to 5) 
 */
```

apply below, too

Reply:
for multi line comments only, keep one-liner as is 😉

Reply:
Okay, should I also add a new line infront of a multi line comment then, like
```
/*
 * Init Magnetometer
 * ...
 */
```